### DISCUSSION TITLE
I forked to patch it with `explicitFolding.langs` ...

### DISCUSSION THREAD

**Discussion by smallscript** [zokugun/vscode-explicit-folding#56] [Issue Triage]
I forked to patch it with `explicitFolding.langs` property.

## 0.14.9
- fix conflicts with default Textmate `FoldingRangeProvider` by introducing
  `explicitFolding.langs` to replace `["*"]` hook with explicit language set.
  
In your settings:

```json
"explicitFolding.langs": ["javascript"],
```

Happy to provide source/pull-request as appropriate. My main reason for this patch was that when enabled it hooked FoldingRangeProvider for all languages which broke simple Textmate providers.

I only wanted to use it on *JavaScript* and maybe a few other languages so I could have an alternate region pattern as follows:

```javascript
//{ my collapsible
//}
```
OR the `/*..*/` variant because `//` providers are aggregated which ends up coalescing sections with `//`
```javascript
/*{ my collapsible */
/*} my collapsible (repeated to make it easy to find the end) */
```
I.e., this *collapse-alls* into a SINGLE FOLDED section with multiple FoldingRangeProvider in VsCode at the moment. Which can be quite annoying to work with. It will unfold into two discrete sections with a little fiddling. Note, that this DOES NOT happen if the FoldingRangeProvider is unified into the language provider (I happen to also write a couple language providers for ESS - another project). Which is how I discovered these issues.
```javascript
//{ my first collapsible
//}
//{ my second collapsible
//}
```

```json
  "explicitFolding.langs": ["javascript"],
  "[javascript]":{
    "explicitFolding.rules":[
      {
        "beginRegex": "(?:\\/\\/{|\\/\\*{)",
        "endRegex": "(?:\\/\\/}|\\/\\*})"
      }
    ]
  },
```

CODE CHANGES:
```typescript
function setup() { // {{{
  $disposable.dispose();
  const provider = new MainProvider();
  // SUPPORT EXPLICIT LANGUAGES REGISTRATION (never `*` unless you want to break Textmate providers)
  const config = vscode.workspace.getConfiguration('explicitFolding')
  const langs: any = config.get('langs');
  for (const scheme of SCHEMES) {
    for(const lang of (langs ? langs : ['*'])) {
      const disposable = vscode.languages.registerFoldingRangeProvider({ language: lang, scheme }, provider);
      $disposable.push(disposable);
    }
  }
  $context!.subscriptions.push($disposable);
} // }}}
```

dsim

**Comment by daiyam** [comment]
<del>Can you tell me which Textmate providers the extension is breaking, at least one of them? With an example? Thx</del>

**Comment by daiyam** [comment]
@smallscript Yep, there is an issue but not due to the extension.

Here what I get:
![Screen Shot 2021-07-03 at 9 56 47 PM](https://user-images.githubusercontent.com/587742/124365625-9a09f800-dc49-11eb-96f1-fac1767cc9cb.png)

On `test.js` and `test3.js`, the folding ranges generated by the extension are correct (see output).
But on `test3.js`, there is a extra folding range because the language provider see the comments separated by an empty line as a comment block and adds the folding range.

If I miss understood your issue, can you provide screenshot or better examples? Also can you open an issue instead? Thx

PS: I've found an issue with indented languages (like Python) since the `v0.14.0` but I don't think it's related...

**Comment by daiyam** [comment]
The issue with indented languages was more older, since `v0.9.0`... The solution, already available in `v0.16.0`, is similar to your proposition (without the required list). Maybe it might fix your problem...
